typing/typeclass.ml

```diff
@@ -592,109 +596,110 @@ let rec class_field self_loc cl_num self_type meths vars arg cf =
     (fun () -> class_field_aux self_loc cl_num self_type meths vars arg cf)
 
 and class_field_aux self_loc cl_num self_type meths vars
-    (val_env, met_env, par_env, fields, concr_meths, warn_vals, inher,
+    (class_env, fields, concr_meths, warn_vals, inher,
      local_meths, local_vals) cf =
   let loc = cf.pcf_loc in
   let mkcf desc =
     { cf_desc = desc; cf_loc = loc; cf_attributes = cf.pcf_attributes }
   in
   match cf.pcf_desc with
     Pcf_inherit (ovf, sparent, super) ->
-      let parent = class_expr cl_num val_env par_env sparent in
+      let parent = class_expr cl_num class_env.val_env class_env.par_env sparent in
       let inher =
         match parent.cl_type with
           Cty_constr (p, tl, _) -> (p, tl) :: inher
         | _ -> inher
       in
       let (cl_sig, concr_meths, warn_vals) =
-        inheritance self_type val_env (Some ovf) concr_meths warn_vals
+        inheritance self_type class_env.val_env (Some ovf) concr_meths warn_vals
           sparent.pcl_loc parent.cl_type
       in
       (* Variables *)
-      let (val_env, met_env, par_env, inh_vars) =
+      let (class_env, inh_vars) =
         Vars.fold
-          (fun lab info (val_env, met_env, par_env, inh_vars) ->
+          (fun lab info (class_env, inh_vars) ->
              let mut, vr, ty = info in
-             let (id, val_env, met_env, par_env) =
-               enter_val cl_num vars true lab mut vr ty val_env met_env par_env
-                 sparent.pcl_loc
+             let (id, class_env) =
+                enter_val cl_num vars true lab mut vr ty class_env.val_env
+                  class_env.met_env class_env.par_env sparent.pcl_loc ;
```

Comment:
could we change `enter_val` to take a `class_env` input, instead of the three arguments seprately?